streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -610,17 +610,32 @@ public void setStreamsUncaughtExceptionHandler(final java.util.function.Consumer
         this.streamsUncaughtExceptionHandler = streamsUncaughtExceptionHandler;
     }
 
-    public void waitOnThreadState(final StreamThread.State targetState) {
+    public boolean waitOnThreadState(final StreamThread.State targetState, long timeoutMs) {
+        if (timeoutMs < 0) {
+            timeoutMs = 0;
+        } else if (timeoutMs == 0) {
+            timeoutMs = Long.MAX_VALUE;
```

Comment:
We definitely shouldn't modify the passed in timeout like this -- a user should be able to pass in `0` to mean "don't block at all". Mysteriously blocking forever when they do so would be pretty weird

Reply:
ah yeah, I had to fix this when I was writing my test